lib/suspenders/actions/strip_comments_action.rb

```diff
@@ -0,0 +1,203 @@
+      #
+      #       config.option1 = :value
+      #     end
+      class StripComments
+        class << self
+          def call(source, parser)
+            buffer = Parser::Source::Buffer.new(nil, source: source)
+            rewriter = Parser::Source::TreeRewriter.new(buffer)
+
+            _, comments = parser.parse_with_comments(buffer)
+
+            comments.each do |comment|
+              strip_comment(comment, buffer, rewriter)
+            end
+
+            rewriter.process
+          end
+
+          private
+
+          def strip_comment(comment, buffer, rewriter)
+            expr = comment.location.expression
+
+            if full_line_comment?(expr)
+              expr = full_line_comment_expr(expr, buffer)
+            end
+
+            rewriter.remove(expr)
+          end
+
+          def full_line_comment_expr(expr, buffer)
+            pos = BackwardStringScanner.beginning_of_line_pos(expr, expr.begin_pos)
+
+            Parser::Source::Range.new(buffer, pos, expr.end_pos + 1)
+          end
+
+          def full_line_comment?(expr)
+            expr.source == expr.source_line.lstrip
+          end
+        end
+      end
+
+      # A tiny, non-stateful backward string scanner somewhat inspired
+      # by Ruby's StringScanner. Ruby's StringScanner is unable to
+      # seek backward on a string.
+      class BackwardStringScanner
+        def self.beginning_of_line_pos(expr, initial_pos)
+          skip_before(expr, initial_pos) { |char| char == "\n" }
+        end
+
+        def self.skip_before(expr, initial_pos, &block)
+          skip_until(expr, initial_pos, -1, &block)
+        end
+
+        def self.skip_until(expr, initial_pos, lookup_inc = 0)
+          pos = initial_pos
+
+          loop do
+            break if pos.zero?
+            char = expr.source_buffer.source[pos + lookup_inc]
+            break if yield(char)
+            pos -= 1
+          end
+
+          pos
+        end
+      end
+
+      # The intent of this class is purely aesthetic: remove leading
+      # newlines inside of code scopes like blocks and begin/end.
+      # Example input:
+      #
+      #     module MyGem
+      #
+      #       MyGem.application.configure do |config|
+      #
+      #         config.option1 = true
+      #
+      #         config.option2 = false
+      #       end
+      #     end
+      #
+      # The output is:
+      #
+      #     module MyGem
+      #       MyGem.application.configure do |config|
+      #         config.option1 = true
+      #
+      #         config.option2 = false
+      #       end
+      #     end
+      class StripLeadingScopeNewlines
+        def self.call(source, parser)
+          buffer = Parser::Source::Buffer.new(nil, source: source)
+          ast = parser.parse(buffer)
+
+          LeadingNewlineStripRewriter.new.rewrite(buffer, ast)
+        end
+
+        class LeadingNewlineStripRewriter < Parser::TreeRewriter
+          def on_module(node)
```

Comment:
There are probably more cases to deal with here if we'd want this to be more generic. Currently, we're dealing with modules, classes, and blocks, which may be enough to cleanly delete comments from the config files.